packages/pluggable-widgets-tools/utils/migration.js

```diff
@@ -0,0 +1,212 @@
+const { join } = require("path");
+const { createInterface } = require("readline");
+const { readJson, writeJson } = require("fs-extra");
+const { execSync } = require("child_process");
+const { red, green, yellow } = require("ansi-colors");
+const { copyFileSync, existsSync, mkdirSync } = require("fs");
+
+let requirePatch = false;
+
+const CheckType = {
+    MAJOR: "major",
+    MINOR: "minor",
+    MAJOR_MINOR: "major_minor"
+};
+
+const dependencies = [
+    { name: "@types/jest", version: "^29.0.0", check: CheckType.MAJOR },
+    { name: "@types/react", version: "~18.0.0", check: CheckType.MAJOR },
+    { name: "@types/react-native", version: "~0.70.0", check: CheckType.MINOR },
+    { name: "@types/react-native-push-notification", version: "8.1.1", check: CheckType.MAJOR_MINOR },
+    { name: "@types/react-dom", version: "~18.0.0", check: CheckType.MAJOR },
+    { name: "@types/react-test-renderer", version: "~18.0.0", check: CheckType.MAJOR },
+    { name: "@react-native-firebase/app", version: "17.3.0", check: CheckType.MAJOR_MINOR },
+    { name: "@react-native-firebase/messaging", version: "17.3.0", check: CheckType.MAJOR_MINOR },
+    {
+        name: "react-native-camera",
+        version: "3.40.0",
+        check: CheckType.MAJOR_MINOR,
+        patch: "react-native-camera+3.40.0.patch"
+    },
+    {
+        name: "react-native-gesture-handler",
+        version: "1.10.3",
+        check: CheckType.MAJOR_MINOR,
+        patch: "react-native-gesture-handler+1.10.3.patch"
+    },
+    { name: "react-native-image-picker", version: "5.0.1", check: CheckType.MAJOR },
+    { name: "react-native-maps", version: "0.31.1", check: CheckType.MAJOR_MINOR },
+    { name: "react-native-progress", version: "^5.0.0", check: CheckType.MAJOR },
+    { name: "react-native-push-notification", version: "8.1.1", check: CheckType.MAJOR_MINOR },
+    { name: "react-native-webview", version: "11.26.1", check: CheckType.MAJOR_MINOR }
+];
+const resolutionsOverrides = [
+    { name: "react", version: "18.2.0", check: CheckType.MAJOR_MINOR },
+    { name: "react-native", version: "0.70.7", check: CheckType.MINOR }
+];
+
+function extractVersions(version) {
+    return version
+        .replace(/^\D+/, "")
+        .split(".")
+        .map(stringValue => Number(stringValue));
+}
+
+async function question(question) {
+    const rl = createInterface({ input: process.stdin, output: process.stdout });
+    return new Promise(resolve =>
+        rl.question(yellow(question), answer => {
+            rl.close();
+            resolve(answer);
+        })
+    );
+}
+
+function getOutdatedDependencies(packageDependencies, listOfNewDependencies = dependencies) {
+    const createReturn = dep => ({
```

Comment:
🎄`versionInfo`?

Reply:
This was also refactored away :)

Reply:
That one I couldn't verify, so I left it for later 😄